modules/services/home-manager-auto-upgrade.nix

```diff
@@ -0,0 +1,60 @@
+{ config, lib, pkgs, ... }:
+
+let
+
+  cfg = config.services.home-manager-auto-upgrade;
+
+in {
+  meta.maintainers = [ lib.maintainers.pinage404 ];
+
+  options = {
+    services.home-manager-auto-upgrade = {
+      enable = lib.mkEnableOption ''
+        Home Manager upgrade
+        Service that run `home-manager switch` periodically with a SystemD's service
```

Comment:
As far as I can tell the module doesn't actually update any channels. It just periodically runs `home-manager switch`.

I'm honestly somewhat skeptical about the value of having this module in the Home Manager repository since I imagine it is a niche use-case and offers just very basic functionality.

It could be expanded to attempt to update channels but since I believe channels are used less and less over time, it is likely that the module would be useless to more and more users over time 😕

Reply:
I planned to make another PRs after this one with [my others customs services](https://gitlab.com/pinage404/dotfiles/-/blob/08990dc4e0336b17fd3b4b3832e14c16c578d674/nixpkgs/service/option/)

I have [one dedicated to upgrade channels](https://gitlab.com/pinage404/dotfiles/-/blob/08990dc4e0336b17fd3b4b3832e14c16c578d674/nixpkgs/service/option/nix_channel_auto_update.nix)

Maybe we should blend them, or maybe should i cancel this PR

What do you prefer ?

Reply:
Blending :+1:

Reply:
Blended :smiley: